src/mscorlib/src/System/String.cs

```diff
@@ -1208,6 +1272,42 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
             }
             return foundCount;
         }
+
+        private struct SplitSeparatorArray
+        {
+            private readonly bool _single;
```

Comment:
When you construct this you could convert a null separator to string.Empty, and I think that would give you the behavior you want?